packages/typescript/base.js

```diff
@@ -0,0 +1,439 @@
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/class-literal-property-style.md
+    "@typescript-eslint/class-literal-property-style": ["error", "fields"],
+
+    // Enforces the usage of Record<K, V> type instead of { [key: K]: V }
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-indexed-object-style.md
+    "@typescript-eslint/consistent-indexed-object-style": ["error", "record"],
+
+    // Enforces consistent usage of type assertions
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-type-assertions.md
+    "@typescript-eslint/consistent-type-assertions": [
+      "error",
+      {
+        assertionStyle: "as",
+        objectLiteralTypeAssertions: "allow-as-parameter",
+      },
+    ],
+
+    // Enforces consistent use of type definitions
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-type-definitions.md
+    "@typescript-eslint/consistent-type-definitions": ["error", "type"],
+
+    // Enforces consistent usage of type exports
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-type-exports.md
+    "@typescript-eslint/consistent-type-exports": [
+      "error",
+      { fixMixedExportsWithInlineTypeSpecifier: true },
+    ],
+
+    // Enforces consistent usage of type imports
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-type-imports.md
+    "@typescript-eslint/consistent-type-imports": [
+      "error",
+      { prefer: "type-imports", disallowTypeAnnotations: true },
+    ],
+
+    // Enforces usage of return type on functions and methods
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/explicit-function-return-type.md
+    "@typescript-eslint/explicit-function-return-type": [
+      "error",
+      {
+        allowExpressions: true,
+        allowTypedFunctionExpressions: true,
+        allowHigherOrderFunctions: true,
+        allowDirectConstAssertionInArrowFunctions: true,
+        allowConciseArrowFunctionExpressionsStartingWithVoid: false,
+      },
+    ],
+
+    // Enforces that class members have access modifiers explicitly specified
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/explicit-member-accessibility.md
+    "@typescript-eslint/explicit-member-accessibility": "off",
+
+    // Enforces that module boundaries (exports) are typed. @typescript-eslint/explicit-function-return-type but for exports only
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/explicit-module-boundary-types.md
+    "@typescript-eslint/explicit-module-boundary-types": "off",
```

Comment:
How come this is disabled?

Reply:
I find this to be the first rule everyone turns off. It has its place in libraries, but perhaps not so much in projects. I can reconsider if you think it should be switched on.